test/test_SUITE.erl

```diff
@@ -0,0 +1,128 @@
+%%
+%% The Initial Developer of the Original Code is VMware, Inc.
+%% Copyright (c) 2007-2012 VMware, Inc.  All rights reserved.
+%%
+
+-module(test_SUITE).
+
+-include_lib("eunit/include/eunit.hrl").
+-include_lib("common_test/include/ct.hrl").
+-include_lib("amqp_client/include/amqp_client.hrl").
+
+-compile(export_all).
+
+all() ->
+    [
+     {group, non_parallel_tests}
+    ].
+
+groups() ->
+    [
+     {non_parallel_tests, [], [
+                               with_backend
+                              ]}
+    ].
+
+%% -------------------------------------------------------------------
+%% Testsuite setup/teardown.
+%% -------------------------------------------------------------------
+%%
+configure_backend(Config) ->
+    rabbit_ct_helpers:merge_app_env(Config,
+                                    {rabbit, [
+                                              {auth_backends,
+                                               [rabbit_auth_backend_internal,
+                                                rabbit_auth_backend_amqp]}]}).
+
+init_per_suite(Config) ->
+    rabbit_ct_helpers:log_environment(),
+    Config1 = rabbit_ct_helpers:set_config(Config, [
+                                                    {rmq_nodename_suffix, ?MODULE}
+                                                   ]),
+
+    rabbit_ct_helpers:run_setup_steps(Config1,
+                                      [ fun configure_backend/1 ] ++
+                                      rabbit_ct_broker_helpers:setup_steps()).
+
+end_per_suite(Config) ->
+    rabbit_ct_helpers:run_teardown_steps(Config,
+                                         rabbit_ct_broker_helpers:teardown_steps()).
+
+init_per_group(_, Config) -> Config.
+
+end_per_group(_, Config) -> Config.
+
+init_per_testcase(with_backend = Testcase, Config) ->
+    PrivDir = ?config(priv_dir, Config),
+    DataDir = ?config(data_dir, Config),
+    BuildDir = filename:join([PrivDir, "build"]),
+    ok = filelib:ensure_dir(filename:join(BuildDir, "dummy")),
+    AmqpPort = rabbit_ct_broker_helpers:get_node_config(Config, 0, tcp_port_amqp),
+    Script = filename:join([DataDir, "run_backend.sh"]),
+    Port = erlang:open_port({spawn, Script}, [use_stdio,
+                                              stderr_to_stdout,
+                                              exit_status,
+                                              {env, [{"BUILD", BuildDir},
+                                                     {"AMQP_PORT", integer_to_list(AmqpPort)}
+                                                    ]}]),
+    Config1 = rabbit_ct_helpers:set_config(Config, {backend_port, Port}),
+    % wait for backend to start
+    timer:sleep(5000),
+    rabbit_ct_helpers:testcase_started(Config1, Testcase).
+
+end_per_testcase(with_backend = Testcase, Config) ->
+    Port = ?config(backend_port, Config),
+    print_port_data(Port, []),
+    {os_pid, OsPid} = erlang:port_info(Port, os_pid),
+    ct:pal("backend os pid ~p", [OsPid]),
```

Comment:
To help limit the amount of output in Concourse and Travis, please set the log level of all messages to `LOW_IMPORTANCE`:
```erlang
ct:pal(?LOW_IMPORTANCE, "backend os pid ~p", [OsPid]),
```